_source/_posts/2022-07-19-angular-security-csrf.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: blog_post
+title: "Protect Your Angular App from Cross-Site Request Forgery"
+author: alisa-duncan
+by: advocate
+communities: [security,javascript]
+description: "Dive into Cross-Site Request Forgery (CSRF) mitigation strategies and how to apply them to your Angular application."
+tags: [security, angular]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+
+Previously, I wrote about web security at a high level and the framework-agnostic ways to increase safety and mitigate vulnerabilities.
+
+|Posts in the SPA web security series|
+| --- |
+| 1. [Defend Your SPA from Security Woes](/blog/2022/07/06/spa-web-security) |
+| 2. [Defend Your SPA from Common Web Attacks](/blog/2022/07/08/spa-web-securty-csrf-xss) |
+| 3. **Protect Your Angular App from Cross-Site Request Forgery** |
+
+Now, I want to dive into the vulnerabilities a little deeper. In this short post, we'll dive into **C**ross-**S**ite **R**equest **F**orgery (CSRF) and look at the built-in help you get when using Angular.
+
+{% include toc.md %}
+
+## Cross-Site Request Forgery (CSRF) protection
+
+In the previous post's overview, you learn about how the attack for CSRF occurs when an agitator uses your active session for a trusted site to perform unauthorized actions. We also learned there's built-in support from browsers to mitigate attacks with `SameSite` attributes on cookies, validating the authenticity of the request on the back-end, and potentially having the front-end send a CSRF token to the back-end for every request.
+
+The mitigation strategies primarily require server-side work, except for that game of CSRF token sending, where the client needs to send the token back in a way the back-end can validate.
+
+![Giphy of a dog picking up mail from the postal worker](https://media.giphy.com/media/V6XBe0GQZSNO0/giphy.gif)
+
+When using CSRF tokens, it's essential to tie the token to the user's session so that the back-end can validate the request. The most common ways are through patterns called [Synchronizer Token Pattern](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#synchronizer-token-pattern) and [Double Submit Cookie](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie).
```

Comment:
```suggestion
When using CSRF tokens, it's essential to tie the token to the user's session so that the back-end can validate the request. The most common ways are through the patterns called [Synchronizer Token Pattern](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#synchronizer-token-pattern) and [Double Submit Cookie](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie).
```